JniBridge.bee.cs

```diff
@@ -388,10 +443,168 @@ static void SetupTestProgramOsx(ToolChain toolchain, NativeProgram staticLib, Co
         np.Libraries.Add(new DynamicLibrary(javaLibDir.Combine("libjvm.dylib")));
         np.CompilerSettings().Add(c => c.WithCustomFlags(new [] {$"-Wl,-rpath,{javaLibDir}"}));
         
-        var destDir = new NPath("build").Combine("osx");
+        var destDir = new NPath("build").Combine(Platform.OSX);
         var config = new NativeProgramConfiguration(codegen, toolchain, false);
         var target = np.SetupSpecificConfiguration(config, config.ToolChain.ExecutableFormat).DeployTo(destDir);
         
-        Backend.Current.AddAliasDependency("build:osx:test", target.Paths);
+        Backend.Current.AddAliasDependency($"build:{Platform.OSX}:test", target.Paths);
+    }
+
+    static NativeProgram SetupTestProgramWindows(ToolChain toolchain, NativeProgram staticLib, CodeGen codegen, NPath generatedFilesDir, Jdk jdk, out NPath targetExecutable)
+    {
+        var np = new NativeProgram("JNIBridgeTests");
+        np.Sources.Add(new NPath("test").Files("*.cpp"));
+        np.IncludeDirectories.Add(generatedFilesDir);
+        np.IncludeDirectories.Add(jdk.JavaHome.Combine("include"));
+        np.IncludeDirectories.Add(jdk.JavaHome.Combine("include", "win32"));
+        np.Libraries.Add(staticLib);
+        var javaLibDir = jdk.JavaHome.Combine("lib");
+        np.Libraries.Add(new StaticLibrary(javaLibDir.Combine("jvm.lib")));
+
+        var destDir = new NPath("build").Combine(Platform.Windows);
+        var config = new NativeProgramConfiguration(codegen, toolchain, false);
+        var target = np.SetupSpecificConfiguration(config, config.ToolChain.ExecutableFormat).DeployTo(destDir);
+
+        var exeName = np.Name + ".exe";
+        var pdbName = np.Name + ".pdb";
+        var server = jdk.JavaHome.Combine("jre/bin/server");
```

Comment:
Do I understand correctly that you copy files into the jdk directory (therefore polluting the jdk installation)? Why is this needed?
By the way, with JDK 11 the jre directory will not exist anymore.

Reply:
I think it copies files in the other direction. Still, why?

Reply:
Blind guesses:
- is JAVA_HOME set when you run tests?
- maybe you need class path entry for Java standard libs?

Reply:
Ok, here's my solution - https://github.com/Unity-Technologies/android-jni-bridge/pull/36/commits/a736cbf8d15db719c4f1af06a0f0dc9900fe6b6b

Don't know about JAVA_HOME, probably not set. But I probably wouldn't want to set it anyways, since it would conflict with my JAVA on PC.

Reply:
So the problem was, that your program could not find jvm.dll, which makes sense. Copying it to other place expectedly did not work, because there's plenty of stuff in Java install and some of it is probably required to run.
But I think you want to set JAVA_HOME to what you actually use. I think that variable might be checked, so you risk problems when you have Java installed on machine and that variable set, but use a different one.

Reply:
On a second thought, if you haven't changed things around, then bee script uses Java from JAVA_HOME and falls back to one from Stevedore, so it's probably fine then.